### ISSUE TITLE
Improve rendering of extension methods in docs

### ISSUE THREAD

**Issue by nicolasstucki** [lampepfl/dotty#10528]
Currently, it is quite hard to read the signatures of extension methods because part of the parameters is not included in the signature. It would be much clearer if the signature would include the extension part.

-------------
Current extension method rendering

<img width="848" alt="Screenshot 2020-11-27 at 10 05 51" src="https://user-images.githubusercontent.com/3648029/100431267-322e7600-3098-11eb-8e92-78fa7fa378b3.png">

----------------
Proposed extension method rendering 
<img width="847" alt="Screenshot 2020-11-27 at 10 05 42" src="https://user-images.githubusercontent.com/3648029/100432024-4cb51f00-3099-11eb-8bf3-33d2bce528d4.png">

**Comment by abgruszecki** [comment]
Yeah, we should consider something like this. We wanted to group extension methods together like they are grouped in the source, but that may not have been the best of ideas after all.

**Comment by nicolasstucki** [comment]
I don't think grouping them is that helpful based on what I've seen.

**Comment by BarkingBad** [comment]
Hi @nicolasstucki,
do you think that we should keep extension part always on top, or maybe do it the same way we handle annotations. Visible only when clicked and expanded?

**Comment by nicolasstucki** [comment]
I think we should always show it as it has important type information about the arguments.

**Comment by nicolasstucki** [comment]
See for example http://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TreeMethods.html. In that case we want to see the extra type parameters or given parameters of the extension in each signature. BTW the type parameters are missing.

**Comment by nicolasstucki** [comment]
An in that case the two identical grouping also show why it is useless to group and factor out the extension part.
